test/e2e/dse_test.go

```diff
@@ -75,6 +76,32 @@ func createSingleDseDatacenterCluster(t *testing.T, ctx context.Context, namespa
 	checkStargateK8cStatusReady(t, f, ctx, kcKey, dcKey)
 }
 
+// createSingleDseDatacenterCluster creates a K8ssandraCluster with one CassandraDatacenter running
+func createSingleHcdDatacenterCluster(t *testing.T, ctx context.Context, namespace string, f *framework.E2eFramework) {
+	t.Log("check that the K8ssandraCluster was created")
+	kc := &api.K8ssandraCluster{}
+	kcKey := types.NamespacedName{Namespace: namespace, Name: "test"}
+	err := f.Client.Get(ctx, kcKey, kc)
+	require.NoError(t, err, "failed to get K8ssandraCluster in namespace %s", namespace)
+	dcKey := framework.ClusterKey{K8sContext: f.DataPlaneContexts[0], NamespacedName: types.NamespacedName{Namespace: namespace, Name: "dc1"}}
+	checkDatacenterReady(t, ctx, dcKey, f)
+	checkDatacenterHasHeapSizeSet(t, ctx, dcKey, f)
+	assertCassandraDatacenterK8cStatusReady(ctx, t, f, kcKey, dcKey.Name)
+	dcPrefix := DcPrefix(t, f, dcKey)
+
+	t.Log("Check that we can communicate through CQL with DSE")
+	_, err = f.ExecuteCql(ctx, f.DataPlaneContexts[0], namespace, kc.SanitizedName(), dcPrefix+"-default-sts-0", string(kc.Spec.Cassandra.ServerType),
+		"SELECT * FROM system.local")
+	require.NoError(t, err, "failed to execute CQL query against DSE", err)
+	opts := kubectl.Options{
+		Namespace: namespace,
+		Context:   f.DataPlaneContexts[0],
+	}
+	output, err := kubectl.Exec(opts, dcPrefix+"-default-sts-0", "ps", "-aux")
+	require.NoError(t, err, "failed to execute ps command")
+	assert.Contains(t, output, "java -Dhcd.server_process -Xms536870912 -Xmx536870912", "expected heap size to be set to 512M")
```

Comment:
I'm not sure if this will pass, because the substring `java -Dhcd.server_process -Xms536870912 -Xmx536870912` may not appear continguously in the output, and I'm not sure if that behaviour is always stable. Perhaps assert on each of the substrings independently to avoid flakes?

Reply:
The command line should be built in a predictable way, as long as we don't change the jvm options used in the manifest.